src/formElementPlugins/useDefault.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect } from 'react'
+import { ResponseFull } from '../utils/types'
+import { DEFAULT_LOADING_VALUE } from './ApplicationViewWrapper'
+
+interface UseDefaultProps {
+  defaultValue: any
+  currentResponse: ResponseFull | null
+  loadingValue?: string | null
+  replaceResponseOnDefaultChange?: boolean
+  onChange: (value: any) => void
+  additionalDependencies?: any[]
+}
+
+const useDefault = ({
+  defaultValue,
+  currentResponse,
+  loadingValue = DEFAULT_LOADING_VALUE,
+  replaceResponseOnDefaultChange = true,
+  onChange,
+  additionalDependencies = [],
+}: UseDefaultProps) => {
+  useEffect(() => {
+    if (defaultValue === loadingValue || defaultValue === undefined) return
+
+    if (replaceResponseOnDefaultChange || !currentResponse?.text) onChange(defaultValue)
+  }, [defaultValue, ...additionalDependencies])
```

Comment:
I'm curious, what `...additionalDependencies` does in this context? And why we are observing other additionalDependencies changes? Maybe worth to add a comment on the code!

Reply:
> I'm curious, what `...additionalDependencies` does in this context? And why we are observing other additionalDependencies changes? Maybe worth to add a comment on the code!

For most elements, the response value only needs when the incoming `default` parameter changes. However, for the "Drop-down" plugin, we also need to update it when the `options` parameter changes as well. So you'll note that for that plugin, it also passes `additionalDependencies: [options]` to the `useDefault` hook, so the dependency array for that particular instance will be different to the "standard" one.